File: apps/frontend/src/services/sync/reconnectStore.ts
import type { Side } from '../../types/game';

const RECONNECT_KEY = 'nxtchess:active_game';

interface ActiveGameInfo {
  gameId: string;
  playerColor: Side;
}

export function saveActiveGame(info: ActiveGameInfo): void {
  try {
    window.sessionStorage.setItem(RECONNECT_KEY, JSON.stringify(info));
  } catch {
    /* noop */
  }
}

export function loadActiveGame(): ActiveGameInfo | null {
  try {
    const raw = window.sessionStorage.getItem(RECONNECT_KEY);
    if (raw) return JSON.parse(raw);
  } catch {
    /* noop */
  }
  return null;
}

export function clearActiveGame(): void {
  try {
    window.sessionStorage.removeItem(RECONNECT_KEY);
  } catch {
    /* noop */
  }
}

File: apps/mobile/lib/services/sync/reconnect_store.dart
import 'dart:convert';

import 'package:flutter/foundation.dart';
import 'package:shared_preferences/shared_preferences.dart';
import '../../models/game_types.dart';

const _reconnectKey = 'nxtchess:active_game';

class ActiveGameInfo {
  final String gameId;
  final Side playerColor;

  const ActiveGameInfo({required this.gameId, required this.playerColor});

  Map<String, dynamic> toJson() => {
    'gameId': gameId,
    'playerColor': playerColor == Side.w ? 'w' : 'b',
  };

  factory ActiveGameInfo.fromJson(Map<String, dynamic> json) {
    return ActiveGameInfo(
      gameId: json['gameId'] as String,
      playerColor: json['playerColor'] == 'w' ? Side.w : Side.b,
    );
  }
}

Future<void> saveActiveGame(ActiveGameInfo info) async {
  try {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(_reconnectKey, jsonEncode(info.toJson()));
  } catch (e) {
    if (kDebugMode) debugPrint('reconnect_store error: $e');
  }
}

Future<ActiveGameInfo?> loadActiveGame() async {
  try {
    final prefs = await SharedPreferences.getInstance();
    final raw = prefs.getString(_reconnectKey);
    if (raw != null) {
      return ActiveGameInfo.fromJson(jsonDecode(raw) as Map<String, dynamic>);
    }
  } catch (e) {
    if (kDebugMode) debugPrint('reconnect_store error: $e');
  }
  return null;
}

Future<void> clearActiveGame() async {
  try {
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove(_reconnectKey);
  } catch (e) {
    if (kDebugMode) debugPrint('reconnect_store error: $e');
  }
}
